### PR TITLE
Prefer previous workspace when jumping to library

### LINKED ISSUE #1178: Prefer current workspace when jumping to library code
Currently when jumping into library code not under an active workspace we seem to randomly choose a workspace to open the file in. This causes things to break for gopls since gopls will not work properly for the library package unless it is a dependency of the current workspace.

I propose we change the library workspace selection to prefer the current buffer's workspace if it can't otherwise find a suitable workspace.

### PR DESCRIPTION
**Pull request by psanford**
Remember what the last active workspace(s) was when opening a file in
a library directory. When jumping to a library definition you most
likely want to open that library from the same workspace as your
current buffer.

Fixes #1178

### CODE REVIEW

File: lsp-mode.el [hunk 1]
```diff
@@ -1105,6 +1110,15 @@ to the beginning and ending points in the range correspondingly."
       (cons start end))
      (parent (lsp--find-wrapping-range parent)))))
 
+(defun lsp--focus-buffer (&rest args)
+  "Focus buffer handler.
+
+Remember the last active workspace set so that we can try that first
```
Comment (yyoncho) [inline lsp-mode.el]: typo: workspace -> workspaces.

**Comment by psanford** [comment]
> I wonder what would lead to better behaviour using `lsp--send-request-async` to set `lsp--last-active-workspaces`(using `target-workspaces`) or current solution. WDYT?

Interesting. You are suggesting we keep track based on the last lsp request we issued instead of the last active buffer. That seems like a better solution than advising 'select-window like the current diff does. 

I'll give that a try.

File: lsp-mode.el (hunk 1 above)
Comment (psanford) [reply lsp-mode.el]: Ha. When I used the word "set" I meant in the discrete math sense of the word. I can see how this comment is confusing and will update it.

Comment (yyoncho) [reply lsp-mode.el]: Ah, it was actually me misreading the comment. It is fine to stay as it is.

File: lsp-mode.el [hunk 2]
```diff
@@ -612,6 +612,11 @@ Changes take effect only when a new session is started."
                                         (ess-r-mode . "r"))
   "Language id configuration.")
 
+(defvar lsp--last-active-workspaces nil
+  "Keep track of last active workspace.
+We want to try the last workspace first when jumping into library
+a library directory")
```
Comment (muirdm) [inline lsp-mode.el]: > jumping into library a library directory

```
  paris
  in the
the springtime
```

Comment (psanford) [reply lsp-mode.el]: Fixed.

File: lsp-mode.el (hunk 1 above)
Comment (psanford) [reply lsp-mode.el]: Fixed.

**Comment by psanford** [comment]
I've updated the pr to set the last active workspaces from (lsp--send-request-async).

**Comment by yyoncho** [comment]
Thank you!

**Pull request merged**